### PR TITLE
PR: Wrapper metadata generation.

### PR DESCRIPTION
**Pull request by tuopuu**
This PR implements #34

### CODE REVIEW

File: jyulb/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
-from simphony.engine import ABCEngineExtension
-from simphony.engine import EngineInterface
+from simphony.extension import ABCEngineExtension
+from simphony.extension import EngineInterface
```
Comment (mehdisadeghi) [inline jyulb/__init__.py]: @tuopuu Sorry for the inconvenience, there was a cyclic problem and I had to remove `extension` package. Would you mind  importing from `simphony.engine` again?

Comment (tuopuu) [inline jyulb/__init__.py]: @mehdisadeghi, thanks for pointing it out. I'll try again with simphony.engine

**Comment by mehdisadeghi** [comment]
@tuopuu  I restarted these builds.

**Comment by mehdisadeghi** [comment]
@tuopuu I guess that I have not fixed the cyclic import problem correctly. I'll look at it tomorrow.

**Comment by tuopuu** [comment]
Thanks, @mehdisadeghi. Let's try to fix it tomorrow.

File: jyulb/__init__.py [hunk 2]
```diff
@@ -1,4 +1,66 @@
+from simphony.engine import ABCEngineExtension
+from simphony.engine import EngineInterface
+from simphony.engine.decorators import register
+
+from .internal.isothermal import JYULBEngine as InternalWrapper
```
Comment (mehdisadeghi) [inline jyulb/__init__.py]: I think this import is broken.

File: jyulb/__init__.py [hunk 3]
```diff
@@ -1,4 +1,66 @@
...
+from .fileio.isothermal import JYULBEngine as FileIOWrapper
```
Comment (mehdisadeghi) [inline jyulb/__init__.py]: This one as well.

**Comment by mehdisadeghi** [comment]
@tuopuu Let's see if those imports fix the problem.

**Comment by tuopuu** [comment]
@mehdisadeghi , it seems that there was only a minor problem with use of incorrect module paths in the /jyulb/**init**.py file. Travis test seems to pass now in Github, but it doesn't pass on my laptop which I don't understand. I'll have to find out what's the issue there. Anyway, this PR is ready for review. :)

**Comment by mehdisadeghi** [comment]
@tuopuu Thanks for the fix. Please make sure that it also passes on your laptop, to make sure that everything is all right.

**Comment by tuopuu** [comment]
@mehdisadeghi , this is the error message I'm getting on my laptop. It has Windows/Cygwin, but the same thing happens on Ubuntu on another PC. It seems that the problem is with importing module ´´´domain´´´ that is written in Cython.

# 

## ERROR: jyulb.internal.isothermal.tests.test_plugin_integration (unittest.loader.ModuleImportFailure)

ImportError: Failed to import test module: jyulb.internal.isothermal.tests.test_plugin_integration
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "/usr/lib/python2.7/unittest/loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "simphony-jyulb/jyulb/internal/isothermal/**init**.py", line 1, in <module>
    from jyulb.internal.isothermal import jyulb_engine
  File "simphony-jyulb/jyulb/internal/isothermal/jyulb_engine.py", line 2, in <module>
    from jyulb.internal.common.proxy_lattice import ProxyLattice
  File "simphony-jyulb/jyulb/internal/common/**init**.py", line 1, in <module>
    from jyulb.internal.common import domain
ImportError: cannot import name domain

**Comment by mehdisadeghi** [comment]
@tuopuu I'm working on it.

**Comment by mehdisadeghi** [comment]
@tuopuu I discovered that if you run `setup.py develop` it will run. Still digging the code for the problem (however the main suspect is cyclic import here).

**Comment by mehdisadeghi** [comment]
@tuopuu After through code inspection here is my comment. 

First lets see why the code runs correctly on travis. That is simply because we have used `setup.py develop` in travis.yml file. The next question that follows is why the `setup.py install` fails? As far as I checked, it has most probably nothing to do with cyclic imports. The cython compiled object with `.so` extension is simply missing. That is the reason behind the error. Let me explain better.

When we run `python setup.py develop`, the setup code compiles the cython modules and generates `.so` object and put it in the insallation directory, which is the current directory in this case. There would be only a symbolic link in python packages directory. Running the tests will succeed because `domain.so` or `solver.so` are present.

When we run `python setup.py install` however things are different. The setup code compiles `.so` files and put them in the python library path, which is not any more the current directory. Therefore, when we run tests inside the **source** directory, it fails. Because the `.so` objects do not exist in the source directory.

There are a number of workarounds. Simply using `setup develop` at the moment. Or writing a test function and run tests by calling that function. Copying `.so` files to the source directory before running tests, etc.

**Comment by tuopuu** [comment]
@mehdisadeghi thanks for the explanation. It makes sense to me now. I had no idea the problems could be caused by something like this. Tests went through fine before, and only started failing after the wrapper registeration code was completed. I naturally assumed it had something to do with the new code. Anyway, I'll fix this issue shortly. Probably copying of the .so files is the simplest solution, which should work in future.

**Comment by mehdisadeghi** [comment]
Good job @tuopuu! Apparently .so files are present both in target .egg file and also in source directory. I tried both install and develop methods and both methods work fine.

**Comment by tuopuu** [comment]
Thanks @mehdisadeghi . I think this PR is ready for review.

**Comment by mehdisadeghi** [comment]
I have already reviewed it, it looks fine to me.

**Comment by mehdisadeghi** [comment]
This closes #34

**Pull request merged**